include/pybind11/attr.h

```diff
@@ -544,7 +544,7 @@ template <typename... Extra,
           size_t named = constexpr_sum(std::is_base_of<arg, Extra>::value...),
           size_t self  = constexpr_sum(std::is_same<is_method, Extra>::value...)>
 constexpr bool expected_num_args(size_t nargs, bool has_args, bool has_kwargs) {
-    return named == 0 || (self + named + has_args + has_kwargs) == nargs;
+    return named == 0 || (self + named + (has_args ? 1 : 0) + (has_kwargs ? 1 : 0)) == nargs;
```

Comment:
```suggestion
    return named == 0 || (self + named + int(has_args) + int(has_kwargs)) == nargs;
```
This would be the casting version.

Reply:
`size_t`, before the next compiler comes whining about signed/unsigned operations ;-) (which is actually a valid concern, if you ask me). Is there a difference between `size_t(has_args)` and `static_cast<size_t>(has_args)`? I guess not.

Reply:
I think we allow c-style casts. The latter would be a C++ one, but less readable. I'm not sure I like this better - the if is logically correct, add 1 if true else 0 is logically what we are doing. This and the other assumes that true is 1, which is specified by the standard but still a arbitrary choice, this would not work in bash, for example. ;)

Reply:
OK, agreed :-) I think converting `bool`s to integer types is pretty fine, though. I'd distrust anything that doesn't do this after you ask it to.

Final thing: should this be `size_t` instead of `std::size_t`? I know that's correct, but in the signature, it's also `size_t` without `std::`.

Reply:
Updated.